tests/src/test/java/alluxio/testutils/LocalAlluxioClusterResource.java

```diff
@@ -169,6 +169,8 @@ public void evaluate() throws Throwable {
           }
         } catch (Exception e) {
           throw new RuntimeException(e);
+        } finally {
+          IntegrationTestUtils.releaseMasterPorts();
```

Comment:
You're right that we can leak a port if cluster setup throws an exception, but we can't release ports here because the ports still needs to be allocated during the `statement.evaluate()` below. For symmetry, can we also move the port acquisition to this closure as well?

```java
public void evaluate() throws Throwable {
  IntegrationTestUtils.reserveMasterPorts();
  try {
    ...
  } finally {
    IntegrationTestUtils.releaseMasterPorts();
  }
}

Reply:
If we're throwing the `RuntimeException` on cluster start, wouldn't we never even get to the `statement.evaluate()` anyways?

If true, then we should probably be releasing the ports before we throw the `RuntimeException` anyways.

Reply:
fixed